ci/shared.mill

```diff
@@ -1,8 +1,15 @@
 package build.ci
 
+import coursier.cache.FileCache
+import coursier.util.Artifact
 
-
-def downloadTestRepo(label: String, commit: String, dest: os.Path) = {
-  os.unzip.stream(requests.get.stream(s"https://github.com/$label/archive/$commit.zip"), dest)
+def downloadTestRepo(label: String, commit: String, dest: os.Path)(implicit ctx: mill.api.Ctx.Log) = {
+  val cache = FileCache()
+  // Uncomment when rebootstrapping Mill, to get better logging
+  //   .withLogger(mill.util.CoursierSupport.millLogger())
+  val file = cache.file(Artifact(s"https://github.com/$label/archive/$commit.zip"))
```

Comment:
Noob question: I just can create a file cache and download a arbitrary file?
Where is this then cached and what is its cache key? I assume the key is the URL?

Because this seems very useful in general. I have a 'manual' cache so far.